domain/service/impl/invitation_service.go

```diff
@@ -10,47 +10,71 @@ import (
-	return &InvitationService{invitationStorage: *invitationStorage, userStorage: *userStorage}
+func NewInvitationService(invitationStorage *IInvitationStorage, groupStorage *IGroupStorage) IInvitationService {
+	return &InvitationService{invitationStorage: *invitationStorage, groupStorage: *groupStorage}
 }
 
-func (i InvitationService) CreateGroupInvitation(request GroupInputDTO, groupID UUID) error {
+func (i InvitationService) CreateGroupInvitation(request GroupInvitationDTO) error {
 	// get invites from request
 	invites := request.Invites
-	// TODO: change, wrong implementation, look up how to store association in gorm
 
-	// handle group invitations
+	// handle group invitations for all invitees
 	for _, invitee := range invites {
-		groupInvitation := model.CreateGroupInvitation(groupID, invitee)
+		groupInvitation := model.CreateGroupInvitation(request.GroupID, invitee)
 		err := i.invitationStorage.AddGroupInvitation(groupInvitation)
 		if err != nil {
 			return err
 		}
 
 	}
 
-	/*
-		// add group invitation to all users
-		for _, userID := range invites {
-			err = i.IUserStorage.AddGroupInvitation(groupInvitation, userID)
-			// TODO: error handling, should return to which users the invitation could not be added
-			core.LogError(err)
-		}
+	return nil
+}
 
-	*/
+func (i InvitationService) GetGroupInvitationByID(id UUID) (GroupInvitationOutputDTO, error) {
+	group, err := i.invitationStorage.GetGroupInvitationByID(id)
+	if err != nil {
+		return GroupInvitationOutputDTO{}, err
+	}
+	return ToGroupInvitationDTO(group), nil
+}
 
-	return nil
+func (i InvitationService) GetGroupInvitationsFromUser(id UUID) ([]GroupInvitationOutputDTO, error) {
+	groupInvitations, err := i.invitationStorage.GetGroupInvitationsByUserID(id)
+	if err != nil {
+		return nil, err
+	}
+	// convert group invitations to data transfer objects
+	var result []GroupInvitationOutputDTO
+	for _, groupInvitation := range groupInvitations {
+		result = append(result, ToGroupInvitationDTO(groupInvitation))
+	}
+	return result, nil
 }
 
-func (i InvitationService) AcceptGroupInvitation(invitation UUID, userID UUID) error {
-	//TODO implement me
-	panic("implement me")
+// TODO: Handle consistency. AddMemberToGroup and DeleteGroupInvitation should be performed both or none.
+func (i InvitationService) AcceptGroupInvitation(invitationID UUID, userID UUID) error {
+	// TODO: is this really necessary?
+	// get group from invitation
+	invitation, err := i.invitationStorage.GetGroupInvitationByID(invitationID)
+	if err != nil {
+		return err
+	}
+	groupID := invitation.Group.ID
+	// add user to group
+	err = i.groupStorage.AddMemberToGroup(userID, groupID)
+	if err != nil {
+		return err
+	}
+	// delete invitation
+	err = i.invitationStorage.DeleteGroupInvitation(invitationID)
+	return err
 }
 
 func (i InvitationService) DeclineGroupInvitation(invitation UUID, userID UUID) error {
```

Comment:
userID is not used here -> but I assume it's needed for authorization (that still has to be implemented)?

Reply:
I think you are right we can delete the param userID. Most likely, authorization will take place in another place.